apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js

```diff
@@ -76,6 +74,457 @@ describe('WorkshopForm test', () => {
     assert(someControl.exists());
   });
 
+  it('creates and publishes new workshop form', () => {
+    const onPublish = sinon.spy();
+
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter>
+          <WorkshopForm
+            permission={new Permission([WorkshopAdmin])}
+            facilitatorCourses={[]}
+            onSaved={onPublish}
+            today={getFakeToday(false)}
+            readOnly={false}
+          />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    // Set fields required for publishing
+    const locationField = wrapper.find('#location_name').first();
+    locationField.simulate('change', {
+      target: {name: 'location_name', value: 'Test location'},
+    });
+
+    const capacityField = wrapper.find('#capacity').first();
+    capacityField.simulate('change', {
+      target: {name: 'capacity', value: 10},
+    });
+
+    const courseField = wrapper.find('#course').first();
+    courseField.simulate('change', {
+      target: {name: 'course', value: 'Admin/Counselor Workshop'},
+    });
+
+    const subjectField = wrapper.find('#subject').first();
+    subjectField.simulate('change', {
+      target: {name: 'subject', value: 'SLP Intro'},
+    });
+
+    // Publish workshop
+    const publishButton = wrapper.find('#workshop-form-save-btn').first();
+    publishButton.simulate('click');
+
+    expect(onPublish).to.have.been.calledOnce;
```

Comment:
looks like a good compromise to keep the tests simple.

nit: please also check  that onPublish has not been called yet before the publish button is clicked, here and below for onSave